### PR TITLE
[CON-381] Migrate `StoreHoursList` to TypeScript

### PR DESCRIPTION
**Pull request by beau-charman**
Addresses https://aesoponline.atlassian.net/browse/CON-381

### CODE REVIEW

File: src/components/StoreHoursList/StoreHoursList.stories.mdx [hunk 1]
```diff
@@ -9,11 +10,38 @@ import { StoreHoursListFixture } from './StoreHoursList.fixture';
 
 <Canvas>
   <Story name="Base component">
-    <StoreHoursList
-      alternateHoursNote={StoreHoursListFixture.alternateHoursNote}
-      heading={StoreHoursListFixture.heading}
-      hoursList={StoreHoursListFixture.hoursList}
-    />
+    {(() => {
```
Comment (MkMan) [inline src/components/StoreHoursList/StoreHoursList.stories.mdx:13]: I've seen this change in a couple of the PRs but I don't completely understand it. Can you please explain it to me 🙏 😃 ?

**Review by MkMan — APPROVED** [review]

**Review by zack76 — APPROVED** [review]

File: src/components/StoreHoursList/StoreHoursList.stories.mdx (hunk 1 above)
Comment (beau-charman) [reply src/components/StoreHoursList/StoreHoursList.stories.mdx:13]: its so I can set up variables and use regular javascript within a MDX story

Comment (beau-charman) [reply src/components/StoreHoursList/StoreHoursList.stories.mdx:13]: you can't add just a block scope and have multiple statement, that will error - an IIFE allows me to add multiple statements within a single block

Comment (MkMan) [reply src/components/StoreHoursList/StoreHoursList.stories.mdx:13]: Ahhhhh yes, it's an iife. I couldn't make it out in the diff. Awesome, makes sense. Thanks :)

**Pull request merged**